### PR TITLE
CI: Run clj-kondo linter

### PR DESCRIPTION
**Pull request by ariya**
This implements issue #14808. Running time is really fast, < 1 minute.

It will appear as an additional line item in the (GitHub) Actions panel:

image

### CODE REVIEW

File: lint-config.edn [hunk 1]
```diff
@@ -0,0 +1,9 @@
+^:replace
```
Comment (ariya) [inline lint-config.edn:1]: I adopted the config filename and content as mentioned in #14808, feel free to suggest any improvements.

**Comment by dpsutton** [comment]
from `project.clj` i think we also need the following paths on there: `"backend/mbql/src" "shared/src"`. Perhaps we can add this command to the README so that people know where to go to be able to run this linter locally so that CI isn't a surprise.

**Comment by ariya** [comment]
> from `project.clj` i think we also need the following paths on there: `"backend/mbql/src" "shared/src"`. Perhaps we can add this command to the README so that people know where to go to be able to run this linter locally so that CI isn't a surprise.

At this rate, is it easier to just lint all `*.clj` and supply an exception list (e.g. for `test`), if clj-kondo supports that at all?

**Comment by dpsutton** [comment]
I'm not sure. I think these classpath roots are pretty static so i'm not worried about it going out of sync. 

Also, does this linter run green at the moment? If not I can probably fixup its complaints in this PR this afternoon and that way we won't start blocking builds

**Comment by ariya** [comment]
> I'm not sure. I think these classpath roots are pretty static so i'm not worried about it going out of sync.

Acknowledged.

> Also, does this linter run green at the moment? If not I can probably fixup its complaints in this PR this afternoon and that way we won't start blocking builds

If "green" means that the exit code doesn't indicate any failures, yes, that is the case right now.

There are outputs, but those are just warnings. Never hurt to clean-up the warnings :-)

**Comment by dpsutton** [comment]
Oh I figured the presence of these warnings would make this fail. I figured this should give us the all clear otherwise report that there are things needing fixing. Same as the docstring checker, etc.

**Review by camsaul — APPROVED** [review]
We should probably change some of these warnings to errors, otherwise it's going to always be green no matter what. 

But we can always merge this in and then tweak the config and fix anything after the fact. LGTM either way

**Review by dpsutton — APPROVED** [review]

**Pull request merged**